### PR TITLE
Fix XCode 11.7 clang errors

### PR DESCRIPTION
**Pull request by andyp1per**
Clang got more picky

### CODE REVIEW

File: libraries/GCS_MAVLink/GCS_Common.cpp [hunk 1]
```diff
@@ -2152,7 +2152,7 @@ MAV_RESULT GCS_MAVLINK::_set_mode_common(const MAV_MODE _base_mode, const uint32
 {
     MAV_RESULT result = MAV_RESULT_UNSUPPORTED;
     // only accept custom modes because there is no easy mapping from Mavlink flight modes to AC flight modes
-    if (_base_mode & MAV_MODE_FLAG_CUSTOM_MODE_ENABLED) {
+    if (uint32_t(_base_mode) & MAV_MODE_FLAG_CUSTOM_MODE_ENABLED) {
```
Comment (khancyr) [inline libraries/GCS_MAVLink/GCS_Common.cpp:2155]: nitpick : uint16_t should be enough !

**Review by khancyr — APPROVED** [review]
good find ! 
the issue with the const on the ringbuffer is weird ... I wonder why it doesn't fail to compile with GCC...

**Comment by peterbarker** [comment]
@WickedShell are we still really, really trying to avoid changing the LUA source?

**Comment by andyp1per** [comment]
@peterbarker it's either change the source or change the compile options. -Werror has to go if we don't want to change the source.

**Review by tridge — APPROVED** [review]

**Comment by tridge** [comment]
looks fine to me, but we will see what @WickedShell thinks

**Comment by peterbarker** [comment]
@WickedShell would prefer we modify the flags we use when compiling the LUA source. 

Precedent is here, apparently: https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_GPS/wscript

 @andyp1per are you up to doing those modifications?

**Comment by andyp1per** [comment]
Mods done, warnings squashed

**Comment by andyp1per** [comment]
@peterbarker @tridge this works but the CFLAGS leak into the rest of the build. Fixing this is beyond me I am afraid.

**Review by WickedShell — APPROVED** [review]
I'm fine with the scripting changes as this at least doesn't make the existing conflict situation any worse. Ideally we take the time in the future to do a real wscript for the lua/src folder, but till then this works.

File: libraries/AP_Scripting/lua_boxed_numerics.cpp [hunk 2]
```diff
@@ -26,7 +26,7 @@ uint32_t coerce_to_uint32_t(lua_State *L, int arg) {
     { // float
         int success;
         const lua_Number v = lua_tonumberx(L, arg, &success);
-        if (success && v >= 0 && v <= UINT32_MAX) {
+        if (success && v >= 0 && v <= float(UINT32_MAX)) {
```
Comment (tridge) [inline libraries/AP_Scripting/lua_boxed_numerics.cpp:29]: we need to check this really works, especially for values close to UINT32_MAX. Can you please check that coercing UINT32_MAX doesn't produce an error?

**Review by tridge — CHANGES_REQUESTED** [review]

File: libraries/AP_Scripting/lua_boxed_numerics.cpp (hunk 2 above)
Comment (tridge) [reply libraries/AP_Scripting/lua_boxed_numerics.cpp:29]: @WickedShell says he is happy with this

**Pull request merged**